subctl/issue-196-disbale-features-in-build.md

```diff
@@ -0,0 +1,88 @@
+# Disable certain features while building `subctl`
+
+Related Issue:
+[Support disabling certain features in builds](https://github.com/submariner-io/enhancements/issues/196)
+
+## Summary
+
+Some builds of `subctl` (e.g. for OCM) don’t need certain sub-commands such as `deploy-broker`, `join` etc. It would be useful to be able
+to disable these features so that users don’t even see them in `--help` output.
+
+## Proposal
+
+Go provides [build constraints](https://pkg.go.dev/cmd/go#hdr-Build_constraints) to identify files that should be included in the package.
+Build constraints are given by a line comment that begins
+
+    //go:build
+
+Constraints may appear in any kind of source file (not just Go), but they must appear near the top of the file, preceded only by blank
+lines and other line comments. These rules mean that in Go files a build constraint must appear before the package clause.
+
+To distinguish build constraints from package documentation, a build constraint should be followed by a blank line.
+
+A build constraint comment is evaluated as an expression containing build tags combined by ||, &&, and ! operators and parentheses.
+Operators have the same meaning as in Go.
```

Comment:
This is copied from https://pkg.go.dev/cmd/go#hdr-Build_constraints; I don’t think it’s particularly useful to do so in this proposal. Instead, the proposal should explain how the feature will be used in Submariner:

```suggestion
Go provides [build constraints](https://pkg.go.dev/cmd/go#hdr-Build_constraints) which can be used to include or exclude files. In this proposal, these would be used to exclude files which are unnecessary for certain features.

We can exclude specific commands by adding `//go:build` constraints to the relevant `cmd` files, with a negated tag.
```

Reply:
Added a `Background` section to explain about build constraints and how to use it.